src/server/server.ts

```diff
@@ -88,6 +88,7 @@ type ExtractStructuredContent<T> = T extends { structuredContent: infer SC }
  *
  * Inspired by tRPC's _def pattern and Hono's type markers.
  */
+// biome-ignore lint/complexity/noBannedTypes: Required for proper type inference and literal type preservation
```

Comment:
all uses of `McpServerTypes` always specify a type - you can remove the default generic type altogether here :)